config/tags.php

```diff
@@ -236,23 +239,117 @@
      */
     'video' => [
         'attr' => [
-            'class',
+            'autoplay',
             'caption',
+            'controls',
+            'class',
             'height',
-            'width'
+            'loop',
+            'muted',
+            'poster',
+            'preload',
+            'style',
+            'width',
         ],
         'html' => function ($tag) {
-            $video = Html::video(
-                $tag->value,
-                $tag->kirby()->option('kirbytext.video.options', []),
-                [
-                    'height' => $tag->height ?? $tag->kirby()->option('kirbytext.video.height'),
-                    'width'  => $tag->width  ?? $tag->kirby()->option('kirbytext.video.width'),
-                ]
-            );
-
-            return Html::figure([$video], $tag->caption, [
-                'class' => $tag->class  ?? $tag->kirby()->option('kirbytext.video.class', 'video'),
+            // all available video tag attributes
+            $availableAttrs = KirbyTag::$types[$tag->type]['attr'];
+
+            // global video tag options
+            $attrs   = $tag->kirby()->option('kirbytext.video', []);
+            $options = $attrs['options'] ?? [];
+
+            // removes options from attributes
+            if (isset($attrs['options']) === true) {
+                unset($attrs['options']);
+            }
+
+            // injects default values from global options
+            // applies only defined attributes to safely update tag props
+            foreach ($attrs as $key => $value) {
+                if (
+                    in_array($key, $availableAttrs) === true &&
+                    (isset($tag->{$key}) === false || $tag->{$key} === null)
+                ) {
+                    $tag->{$key} = $value;
+                }
+            }
+
+            // generates tag attributes helper inline function
+            // for ex: `autoplay` will not work if `false` is a `string` instead of a `boolean`
+            // this is why the type converter is used in function
+            $generateTagAttributes = function ($tag) {
```

Comment:
Do we need an inline function here if we only call it once? Could be just the code, without putting it in a function, or?

Reply:
You are absolutely right. Previously, I was calling this inline function from two places. So I needed an inline function. After improvement I used it in one place but forgot it there 🙈

Reply:
Done